src/frontend/komponenter/Fagsak/Vedtak/VedtakBegrunnelserTabell/Context/VedtaksbegrunnelseTeksterContext.tsx

```diff
@@ -4,24 +4,69 @@ import constate from 'constate';
 
 import { useHttp } from '@navikt/familie-http';
 import { Ressurs, byggTomRessurs } from '@navikt/familie-typer';
+import { byggSuksessRessurs } from '@navikt/familie-typer/dist/ressurs';
 
+import { useSanity, Begrunnelsedata } from '../../../../../api/sanity/sanityHook';
+import { sanityApiNavnTilBegrunnelseDictionary } from '../../../../../api/sanity/typer';
+import { useApp } from '../../../../../context/AppContext';
+import { ToggleNavn } from '../../../../../typer/toggles';
+import { VedtakBegrunnelseType } from '../../../../../typer/vedtak';
 import { VedtaksbegrunnelseTekster } from '../../../../../typer/vilkår';
 
 const [VedtaksbegrunnelseTeksterProvider, useVedtaksbegrunnelseTekster] = constate(() => {
     const { request } = useHttp();
+    const { toggles } = useApp();
+    const { hentBegrunnelser } = useSanity();
 
     const [vedtaksbegrunnelseTekster, settVedtaksbegrunnelseTekster] = React.useState<
         Ressurs<VedtaksbegrunnelseTekster>
     >(byggTomRessurs());
 
+    const byggVedtaksbegrunnelsesteksterFraSanitydata = (
+        begrunnelsedataFraSanity: Begrunnelsedata[]
+    ) =>
+        begrunnelsedataFraSanity.reduce(
+            (
+                acc: VedtaksbegrunnelseTekster,
+                begrunnelseMetadata: Begrunnelsedata
+            ): VedtaksbegrunnelseTekster => {
+                // Løser dette kun for "Norsk, nordisk bosatt i Norge"-begrunnelsen for POCen
+                begrunnelseMetadata.apiNavn === 'norskNordiskBosattINorge' &&
+                    sanityApiNavnTilBegrunnelseDictionary[begrunnelseMetadata.apiNavn] &&
+                    acc[begrunnelseMetadata.begrunnelsetype].push({
+                        id: sanityApiNavnTilBegrunnelseDictionary[begrunnelseMetadata.apiNavn],
+                        navn: begrunnelseMetadata.navnISystem,
+                        vilkår: begrunnelseMetadata.vilkår,
+                    });
+                return acc;
```

Comment:
```suggestion
                if (begrunnelseMetadata.apiNavn === 'norskNordiskBosattINorge' &&
                    sanityApiNavnTilBegrunnelseDictionary[begrunnelseMetadata.apiNavn]) {
                    return {
                    ...acc,
                    [begrunnelseMetadata.begrunnelsetype]: {
                        id: sanityApiNavnTilBegrunnelseDictionary[begrunnelseMetadata.apiNavn],
                        navn: begrunnelseMetadata.navnISystem,
                        vilkår: begrunnelseMetadata.vilkår,
                    }}
                    } else return acc;
```

Bedre å lage nye objekter fremfor å mutere på acc :)